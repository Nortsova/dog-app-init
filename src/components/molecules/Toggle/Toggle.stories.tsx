import type { Meta, StoryObj } from '@storybook/react';
import { Toggle } from '.';

const meta = {
  title: 'Molecules/Toggle',
  component: Toggle
} satisfies Meta<typeof Toggle>;

export default meta;
type Story = StoryObj<typeof meta>;

export const ToggleComponent: Story = {
  args: {}
};
