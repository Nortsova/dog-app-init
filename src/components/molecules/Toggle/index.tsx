import { Button } from '@mui/material';
import { useState } from 'react';

export const Toggle = () => {
  const [toggle, setToggle] = useState(false);
  const handleToggle = () => setToggle(prev => !prev);
  return (
    <>
      <Button data-testid="toggle-btn" onClick={handleToggle}>
        Toggle
      </Button>
      {toggle && <div data-testid="expanded">expanded</div>}
    </>
  );
};
