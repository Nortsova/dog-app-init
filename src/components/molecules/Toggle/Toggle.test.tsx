import { composeStories } from '@storybook/react';
import { fireEvent, render } from '@testing-library/react';
import * as stories from './Toggle.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('Toggle Component', () => {
  const { ToggleComponent } = composeStories(stories);
  testSnapshot(<ToggleComponent />);

  it('Should expand toggle by click', () => {
    const { getByTestId, queryByTestId } = render(<ToggleComponent />);
    expect(queryByTestId('expanded')).not.toBeInTheDocument();
    const toggle = getByTestId('toggle-btn');
    fireEvent.click(toggle);
    expect(getByTestId('expanded')).toBeInTheDocument();
  });
});
