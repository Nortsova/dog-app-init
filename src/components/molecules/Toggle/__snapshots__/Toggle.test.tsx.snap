// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Component should match snapshot Component 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: 'Dosis',sans-serif;
  font-weight: 500;
  font-size: 1.1607142857142858rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 4px 5px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #921FED;
  border-radius: 20px;
  box-shadow: none;
  text-transform: none;
  font-size: 20px;
  line-height: 1.6;
  padding: 8px 16px;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(146, 31, 237, 0.04);
}

@media (hover: none) {
  .emotion-0:hover {
    background-color: transparent;
  }
}

.emotion-0.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-0:hover,
.emotion-0:active {
  box-shadow: none;
}

<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall emotion-0"
    data-testid="toggle-btn"
    tabindex="0"
    type="button"
  >
    Toggle
  </button>
</div>
`;
